docs/devguide/flashfirmware.rst

```diff
@@ -20,39 +36,72 @@ Start the build with either yotta::
 
   yt build
 
-...or use the Makefile::
+...or use the makefile::
 
   make all
 
-The result is a microbit-micropython hex file (i.e. ``microbit-micropython.hex``)
-found in the build/bbc-microbit-classic-gcc-nosd/source from the root of the
-repository.
+The result is a microbit-micropython hex file
+(i.e. ``microbit-micropython.hex``) found in the
+build/bbc-microbit-classic-gcc-nosd/source from the root of the repository.
```

Comment:
I think the readme lists `build/firmware.hex`, could you check what's the right path for the final hex?

Reply:
I can't get the build working on my machine at the moment I get this error when trying to set the target
```
dyld: Library not loaded: @executable_path/../.Python
  Referenced from: /Users/mark/yotta-venv/bin/python3.7
  Reason: image not found
[1]    2649 abort      yt target 
```
I will update to `firmware.hex` in line with the readme.

Reply:
Sounds like the yotta virtual environment might have been created from a Python installation that no longer exists (or has changed)?
I assume you would get a similar issue when trying to execute any yotta command?
If so, you'll have to recreate the virtual environment with a new one.

Reply:
Thanks, Carlos. Recreating the virtual environment worked and output is `firmware.hex`

Reply:
And the file is located in `build/bbc-microbit-classic-gcc-nosd/source` is not in `build` ?

Reply:
I get another build error

```
Linking CXX executable source/microbit-micropython
tools/adduicr.py build/bbc-microbit-classic-gcc-nosd/source/microbit-micropython.hex $(cat build/veraddr.txt) -o build/firmware.hex
usage: adduicr.py [-h] [-o OUTPUT] firmware address
adduicr.py: error: argument address: invalid <lambda> value: '0x'
make: *** [build/firmware.hex] Error 2
```
but I assume the output is supposed to be at `build/firmware.hex`

Reply:
Yeah, I'm getting the same issue on my normal dev environment, but not on an old VM. The source code hasn't changed so it might be related to using newer toolchain.
I can confirm the location, can you update the docs and we'll merge the PR?

Reply:
Is the original text not correct then?


This is still true as a `microbit-micropython.hex` is created in `build/bbc-microbit-classic-gcc-nosd/source`
```
The result is a microbit-micropython hex file
(i.e. ``microbit-micropython.hex``) found in the
build/bbc-microbit-classic-gcc-nosd/source from the root of the repository.
```
As is this as a final 'firmware.hex` file is created in /build
```
The Makefile does some extra preprocessing of the source, which is needed
only if you add new interned strings to qstrdefsport.h. The Makefile also puts
the resulting firmware at build/firmware.hex, and includes some convenience
targets.
```

Reply:
Yeah, the first block is correct, but mentioning the `microbit-micropython.hex` file in the build instructions can be accidentally interpreted as that being the main hex file to use (the text does **not** say that, but it's common for users to skim over the docs to find "the bit they need" even if that ends up being the wrong bit).
I'd rather remove mentions of the "middle steps" hex files to avoid people accidentally using that instead of the final hex.


>The Makefile does some extra preprocessing of the source, which is needed
only if you add new interned strings to qstrdefsport.h. The Makefile also puts
the resulting firmware at build/firmware.hex, and includes some convenience
targets.

The extra preprocessing now also includes adding the UICR, which is needed for the final hex, so using the makefile is always required now.

Reply:
Thanks, Carlos. Agreed. I have removed the unnecessary sentence and updated in https://github.com/bbcmicrobit/micropython/pull/711/commits/7d6951b3d59d0b28c472e7a88c2e2302468d0032

Reply:
I've opened https://github.com/bbcmicrobit/micropython/issues/712 about the build issues